holoviews/core/dimension.py

```diff
@@ -1077,13 +1077,25 @@ def __unicode__(self):
         return unicode(PrettyPrinter.pprint(self))
 
 
-
     def __call__(self, options=None, **kwargs):
+        self.warning('Use opts method instead')
```

Comment:
We can do the following things:

1. Document using opts only (no argument there!)
2. Mention that people should please stop using ``__call__`` in the release notes where we mention the addition of opts.
3. When we do enable the warning, you should be able to disable it via the config system we really want and need (set deprecation settings, set styles, other things like that!)